src/goto-symex/renaming_level.cpp

```diff
@@ -67,6 +67,8 @@ operator()(renamedt<ssa_exprt, L0> l0_expr) const
 renamedt<ssa_exprt, L2> symex_level2t::
 operator()(renamedt<ssa_exprt, L1> l1_expr) const
 {
+  if(!l1_expr.get().get_level_2().empty())
+    return renamedt<ssa_exprt, L2>{std::move(l1_expr.value)};
```

Comment:
This change seems surprising to me, but I'm likely just missing something. Would you mind explaining?

Reply:
As for the symex_level1t case, it does not change the behaviour because it is only called from set_l2_indices which had the same check until this commit. We need the check to be here and not in `set_l2_indices`, because `set_l2_indices` cannot create `renamedt` expression, the constructor is private, this ensures we cannot make a `renamedt` without checking it has the right indices set.